crates/nu-command/src/filters/take/take_.rs

```diff
@@ -78,99 +124,26 @@ impl Command for Take {
                     span: Span::test_data(),
                 }),
             },
+            Example {
+                description: "Return the first 2 bytes of a binary value",
+                example: "0x[01 23 45] | take 2",
+                result: Some(Value::Binary {
+                    val: vec![0x01, 0x23],
+                    span: Span::test_data(),
+                }),
+            },
+            Example {
+                description: "Return the first 3 elements of a range",
+                example: "1..10 | take 3",
+                result: Some(Value::List {
+                    vals: vec![Value::test_int(1), Value::test_int(2), Value::test_int(3)],
+                    span: Span::test_data(),
+                }),
+            },
         ]
     }
 }
 
-fn first_helper(
-    engine_state: &EngineState,
-    stack: &mut Stack,
-    call: &Call,
-    input: PipelineData,
-) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
-    let head = call.head;
-    let mut rows_desired: usize = call.req(engine_state, stack, 0)?;
-
-    let ctrlc = engine_state.ctrlc.clone();
-    let metadata = input.metadata();
-
-    let mut input_peek = input.into_iter().peekable();
-    if input_peek.peek().is_some() {
-        match input_peek
-            .peek()
-            .ok_or_else(|| {
-                ShellError::GenericError(
-                    "Error in first".into(),
-                    "unable to pick on next value".into(),
-                    Some(call.head),
-                    None,
-                    Vec::new(),
-                )
-            })?
-            .get_type()
-        {
-            Type::Binary => {
-                match &mut input_peek.next() {
-                    Some(v) => match &v {
-                        Value::Binary { val, .. } => {
-                            let bytes = val;
-                            if bytes.len() >= rows_desired {
-                                // We only want to see a certain amount of the binary
-                                // so let's grab those parts
-                                let output_bytes = bytes[0..rows_desired].to_vec();
-                                Ok(Value::Binary {
-                                    val: output_bytes,
-                                    span: head,
-                                }
-                                .into_pipeline_data())
-                            } else {
-                                // if we want more rows that the current chunk size (8192)
-                                // we must gradually get bigger chunks while testing
-                                // if it's within the requested rows_desired size
```

Comment:
For posterity: I think this code was left over from an earlier time and no longer necessary. I tested the new functionality and it is able to get byte ranges larger than 8192 bytes.